core/memory_tracker/cc/memory_tracker.h

```diff
@@ -311,47 +297,48 @@ class DirtyPageTable {
       current_;  // The space for the last page address stored.
 };
 
-// MemoryTracker utilizes Segfault signal on Linux to track accesses to
+// MemoryTrackerImpl utilizes Segfault signal on Linux to track accesses to
 // memories.
-class MemoryTracker {
+template<typename SpecificMemoryTracker>
```

Comment:
The style guide suggests naming them the same as Classes.
https://google.github.io/styleguide/cppguide.html#Type_Names

The names of all types — classes, structs, type aliases, enums, and type template parameters — have the same naming convention